addons/account/models/account_payment.py

```diff
@@ -983,6 +985,7 @@ def button_request_cancel(self):
 
     def action_draft(self):
         self.state = 'draft'
+        self.move_id.button_draft()
```

Comment:
Doesn't take into account lock dates/hashes I think (may be worth to add a test)